pkg/state/statefile/statefile.go

```diff
@@ -84,6 +86,78 @@ var ErrInvalidMetadataLength = fmt.Errorf("metadata length invalid, maximum size
 // ErrMetadataInvalid is returned if passed metadata is invalid.
 var ErrMetadataInvalid = fmt.Errorf("metadata invalid, can't start with _")
 
+// ErrInvalidFlags is returned if passed flags set is invalid.
+var ErrInvalidFlags = fmt.Errorf("flags set is invalid")
+
+const (
+	ImageCompressed = "image_compressed"
+)
+
+// Writer options.
+type Flags uint
```

Comment:
It seems weird to have a whole `Flags` bitfield that would presumably carry multiple options for the statefile, yet store it under a metadata key called `image_compressed`.
Also, `metadata` is a string key-value store, so we're not limited to storing raw bits in there.
How about making this be just an `int` or a `string` that represents compression state? Then it can simply be stored as such in `metadata` (with absence of the `image_compressed` key meaning "compressed").

Reply:
yeah. now it became a struct of `statefile.Options` with the `CompressionLevel` string type. Must be better now.